### ISSUE TITLE
RenderFlow Graph second iteration

### ISSUE THREAD

**Issue by RodrigoHolztrattner** [RodrigoHolztrattner/WonderlandProject#1]
Some facts for the second iteration:

- Graph generation (without console outputs) is very fast (less then 1ms) and can be done each logic frame without introducing performance issues.
- ...

For the second iteration, these are the target achievements: 

- The RenderFlow Graph:
  * Must be transformed into "just-a-template-class" that can be easily created and destroyed.
  * It must be able to hold resources used for it's internal render methods(?!).
  * It should do all necessary checks to make sure the types of textures match, and if necessary it must express that a transition is necessary before writing to the backbuffer.
  * Re-use texture templates whenever possible.
  * Accept copying all output attachments to images that can be used as input for other external phases on the overlay system.
- The frame aggregator functionality that creates all GPU objects must be moved to a separated class.
- This class mentioned above must have:
  * A constructor that accept all external allocators, the render graph template(?!) and the current present surface dimensions.
  * All remaining allocators.
  * Support for triple-buffering (or equivalent).
  * A method that will receive the graph template and prepare a list of rendering commands.
  * A method that will execute the internal list of rendering commands for the chosen buffering index.
  * A fast clear method that will make it re-writable every logic frame.
  * Must be able to accept the global hold switch that will be used to lock logic and/or render operations globally.
  * Must be able to handle destruction correctly, waiting for all current operations.

These changes should enable the RenderFlow Graph to be easily created and expressed by nodes on the editor. It will also enable using it for the editor rendering layer, proper allowing the material, special effects and other editors to render it's objects the same way the game will render them.

Basically we will have the RenderFlow Graph and a few other classes representing a group of objects that can handle building all necessary objects for the rendering phase, while also doing it.
